src/cr/cube/matrix.py

```diff
@@ -260,14 +260,17 @@ def _columns_inserted_at_right(self):
     def _iter_columns(self):
         """Generate all column vectors with insertions interleaved at right spot."""
         opposing_insertions = self._all_inserted_rows
+        column_vector_index = 0
 
         # ---subtotals inserted at top---
         for column in self._columns_inserted_at_left:
-            yield _AssembledVector(column, opposing_insertions)
+            yield _AssembledVector(column, opposing_insertions, column_vector_index)
+            column_vector_index += 1
```

Comment:
Better to make the col_idx a sum rather than introduce a stateful variable and mutation:

```python
for col_idx, column in enumerate(self._columns_inserted_at_left):
    yield _AssembledVector(column, opposing_insertions, col_idx)
```

Reply:
Okay, maybe not come to think of it since it needs to fall through ...